plugin/restore/mysql/mysql.go

```diff
@@ -123,6 +132,170 @@ func GetPITRDatabaseName(database string, timestamp int64) string {
+	latest, err := r.showLatestBinlogFile(ctx)
+	if err != nil {
+		return err
+	}
+	return r.syncBinlog(ctx, instance, saveDir, *latest)
+}
+
+// syncBinlog syncs the binlog specified by `meta` between the instance and local.
+func (r *Restore) syncBinlog(ctx context.Context, instance *api.Instance, saveDir string, meta mysql.BinlogFile) error {
+	tmpFilePrefix := fmt.Sprintf("tmp_%d_", time.Now().UnixNano())
+	// TODO(zp): support ssl?
+	cmd := exec.CommandContext(ctx, filepath.Join(mysqlbinlogPath, "bin", "mysqlbinlog"),
+		meta.Name,
+		fmt.Sprintf("--read-from-remote-server --host=%s --port=%s --user=%s --password=%s", instance.Host, instance.Port, instance.Username, instance.Password),
+		"--raw",
+		// Note, recheck here when upgrade embedding mysqlbinlog binary
+		// for mysqlbinlog with binlog.000001 --result-file=a_, it will save it as a_binlog.000001
+		fmt.Sprintf("--result-file=%s", filepath.Join(saveDir, tmpFilePrefix)),
+	)
+	cmd.Stdout = os.Stderr
+	cmd.Stderr = os.Stderr
+
+	var runErr error
+	defer func() {
+		tmpFileName := fmt.Sprintf("%s%s", tmpFilePrefix, meta.Name)
+		tmpFilePath := filepath.Join(saveDir, tmpFileName)
+		if runErr != nil {
+			// delete the download binlog if some error occur
+			_ = os.Remove(tmpFilePath)
+			return
+		}
+
+		fi, err := os.Stat(tmpFilePath)
+		if err != nil {
+			// delete the download binlog if cannot stat it
+			_ = os.Remove(tmpFilePath)
+			return
+		}
+		if fi.Size() != meta.Size {
+			// delete the download binlog if filesize not match
+			_ = os.Remove(tmpFilePath)
+			return
+		}
+
+		// no error, rename it
+		_ = os.Rename(tmpFilePath, filepath.Join(saveDir, meta.Name))
+	}()
+
+	if runErr = cmd.Run(); runErr != nil {
+		return fmt.Errorf("cannot run mysqlbinlog, error: %w", runErr)
+	}
+	return nil
+}
+
+// showBinlogFiles returns the metadata of binlogs
+func (r *Restore) showBinlogFiles(ctx context.Context) ([]mysql.BinlogFile, error) {
+	db, err := r.driver.GetDbConnection(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+	// https://dev.mysql.com/doc/refman/8.0/en/show-binary-logs.html
+	stmt := `SHOW BINARY LOGS;`
+	rows, err := db.QueryContext(ctx, stmt)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var binlogFiles []mysql.BinlogFile
+	for rows.Next() {
+		var binlogFile mysql.BinlogFile
+		var unused interface{}
+		if err := rows.Scan(&binlogFile.Name, &binlogFile.Size, &unused /*Encrypted column*/); err != nil {
+			return nil, err
+		}
+		binlogFiles = append(binlogFiles, binlogFile)
+	}
+	return binlogFiles, nil
+}
+
+// showLatestBinlogFile returns the metadata of latest binlog
+func (r *Restore) showLatestBinlogFile(ctx context.Context) (*mysql.BinlogFile, error) {
+	db, err := r.driver.GetDbConnection(ctx, "")
+	if err != nil {
+		return nil, err
+	}
+
+	stmt := `SHOW MASTER STATUS;`
+	rows, err := db.QueryContext(ctx, stmt)
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	var meta mysql.BinlogFile
```

Comment:
```suggestion
	var binlogFiles []mysql.BinlogFile
```
Should be consistent with other places.